### PR TITLE
Add Power{N} operator in computational graph

### PR DESCRIPTION
**Pull request by houpc**

### CODE REVIEW

**Review by dcerkoney — APPROVED** [review]
Looks good to me

**Comment by houpc** [comment]
> Looks good to me

Thanks. The modification for static compilers has been updated.

File: src/backend/static.jl [hunk 1]
```diff
@@ -1,35 +1,54 @@
...
-        return "(" * join(["g$(g.id)" for g in subgraphs], " * ") * ")"
+        terms = ["g$(g.id)" * (gfactor == 1 ? "" : " * $gfactor") for (g, gfactor) in zip(subgraphs, subgraph_factors)]
+        return "(" * join(terms, " * ") * ")"
+        # return "(" * join(["g$(g.id)" for g in subgraphs], " * ") * ")"
     end
 end
 
+function _to_static(::Type{ComputationalGraphs.Power{N}}, subgraphs::Vector{Graph{F,W}}, subgraph_factors::Vector{F}) where {N,F,W}
+    factor_str = subgraph_factors[1] == 1 ? "" : " * $(subgraph_factors[1])"
+    return "((g$(subgraphs[1].id))^$N$factor_str)"
```
Comment (dcerkoney) [inline src/backend/static.jl:24]: I suggest moving the " * " from `factor_str` to the return statement for readability everywhere in `static.jl`. This way it is clearer at a glance what each expression compiles to. For example,
```julia
factor_str = subgraph_factors[1] == 1 ? "" : "$(subgraph_factors[1])"
return "((g$(subgraphs[1].id))^$N * $factor_str)"
```

**Review by dcerkoney — APPROVED** [review]

File: src/backend/static.jl (hunk 1 above)
Comment (houpc) [reply src/backend/static.jl:24]: Thank you for your suggestion. But "*" should be absent in this expression when subgraphs_factors[1]==1. I changed it this way mainly to reduce one redundant multiplication operation.

Comment (dcerkoney) [reply src/backend/static.jl:24]: Ah, of course, I missed this somehow.

**Pull request merged**